src/main/java/de/hysky/skyblocker/skyblock/dwarven/CorpseFinder.java

```diff
@@ -0,0 +1,270 @@
+        if (SkyblockerConfigManager.get().mining.glacite.enableParsingChatCorpseFinder) parseCords(text);  // parsing cords from chat
+
+        Matcher matcherCorpse = CORPSE_FOUND_PATTERN.matcher(text.getString());
+        if (matcherCorpse.find()) {
+            LOGGER.info(PREFIX + "Triggered code for onChatMessage");
+            LOGGER.warn(PREFIX + "State of corpsesByType: {}", corpsesByType);
+            String corpseType = matcherCorpse.group(1).toUpperCase();
+            String key = switch (corpseType) {   // there is probably less stupid way to do this
+                case ("LAPIS") -> "LAPIS_ARMOR_HELMET";
+                case ("UMBER") -> "ARMOR_OF_YOG_HELMET";
+                case ("TUNGSTEN") -> "MINERAL_HELMET";
+                case ("VANGUARD") -> "VANGUARD_HELMET";
+                default -> "";
+            };
+            if (MinecraftClient.getInstance().player == null) return;
+            List<Corpse> corpses = corpsesByType.get(key);
+            if (corpses == null) {
+                LOGGER.warn(PREFIX + "Couldn't get corpses! corpseType: {}, key: {}", corpseType, key);
+                return;
+            }
+            Corpse closestCorpse = null;
+            float closestDistance = 100;
+            for (Corpse corpse : corpses) {
+                float distance = corpse.entity.distanceTo(MinecraftClient.getInstance().player);
+                if (distance < closestDistance) {
+                    closestDistance = distance;
+                    closestCorpse = corpse;
+                }
+            }
+            if (closestCorpse != null) {
+                LOGGER.info(PREFIX + "Found corpse, marking as found! {}", closestCorpse.entity);
+                closestCorpse.waypoint.setFound();
+                closestCorpse.opened = true;
+            } else LOGGER.warn(PREFIX + "Couldn't find closest corpse despite triggering onChatMessage!");
+        }
+    }
+
+    private static void setSeen(Corpse corpse) {
+        corpse.seen = true;
+        if (System.currentTimeMillis() - corpse.messageLastSent < 300)  return;
+
+        corpse.messageLastSent = System.currentTimeMillis();
+        MinecraftClient.getInstance().player.sendMessage(
+            Constants.PREFIX.get()
+                .append("Found a ")
+                .append(Text.literal(corpse.name + " Corpse")
+                    .withColor(corpse.color.getColorValue()))
+                .append(" at " + corpse.entity.getBlockPos().up(0).toShortString() + "!")
+                .styled(style -> style.withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/skyblocker corpseHelper shareLocation " + PosUtils.toSpaceSeparatedString(corpse.waypoint.pos) + " " + corpse.name))
+                    .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Text.literal("Click to share the location in chat!").formatted(Formatting.GREEN)))));
+    }
+
+    static class Corpse {
```

Comment:
inner classes are usually at the bottom of the file but i'm nitpicking at this point

Reply:
also moved